### PR TITLE
feat(cxl-ui): add ability for main content to be scrolled from the ou…

### PR DESCRIPTION
**Pull request by anoblet**
…tside

### CODE REVIEW

**Comment by anoblet** [comment]
@leho I don't have access to the document :)

File: packages/cxl-ui/src/components/cxl-app-layout.js [hunk 1]
```diff
@@ -118,5 +122,13 @@ export class CXLAppLayoutElement extends LitElement {
     registerGlobalStyles(cxlAppLayoutGlobalStyles, {
       moduleId: 'cxl-app-layout-global',
     });
+
+    this.addEventListener('wheel', this._onWheel.bind(this));
```
Comment (lkraav) [inline packages/cxl-ui/src/components/cxl-app-layout.js:126]: Can we set up and tear down listening with `wide` mode? This doesn't need to listen on mobiles.

**Review by lkraav — CHANGES_REQUESTED** [review]

File: packages/cxl-ui/src/components/cxl-app-layout.js (hunk 1 above)
Comment (anoblet) [reply packages/cxl-ui/src/components/cxl-app-layout.js]: Fixed

File: packages/cxl-ui/src/components/cxl-app-layout.js [hunk 2]
```diff
@@ -1,13 +1,14 @@
...
 import '@conversionxl/cxl-lumo-styles';
 import { registerGlobalStyles } from '@conversionxl/cxl-lumo-styles/src/utils';
-import cxlAppLayoutStyles from '../styles/cxl-app-layout-css.js';
-import cxlAppLayoutGlobalStyles from '../styles/global/cxl-app-layout-css.js';
 import '@vaadin/vaadin-button';
 import '@vaadin/vaadin-context-menu/src/vaadin-device-detector.js';
+import { customElement, html, LitElement, property, query } from 'lit-element';
+import normalizeWheel from 'normalize-wheel';
+import cxlAppLayoutStyles from '../styles/cxl-app-layout-css.js';
+import cxlAppLayoutGlobalStyles from '../styles/global/cxl-app-layout-css.js';
```
Comment (lkraav) [inline packages/cxl-ui/src/components/cxl-app-layout.js:11]: Did ESLint or something auto-re-arrange these?

Ideally we'd only have the relevant import change here.

File: packages/cxl-ui/package.json [hunk 3]
```diff
@@ -24,7 +24,8 @@
     "@vaadin/vaadin-themable-mixin": "^1.6.2",
     "headroom.js": "^0.12.0",
     "lit-element": "^2.2.1",
-    "lit-html": "^2.0.0-rc.3"
+    "lit-html": "^2.0.0-rc.3",
+    "normalize-wheel": "https://github.com/anoblet/normalize-wheel#esm"
```
Comment (lkraav) [inline packages/cxl-ui/package.json:28]: If we have to fork this anyway, perhaps we can also drop some of the general compatibility check code from it? If not, that's fine, too, but worth a look.

File: packages/cxl-ui/src/components/cxl-app-layout.js (hunk 2 above)
Comment (anoblet) [reply packages/cxl-ui/src/components/cxl-app-layout.js:11]: That's a habit of mine: https://code.visualstudio.com/docs/languages/typescript#_organize-imports

I can change it back, here's the relevant eslint rule if we decide to enable it in the future: https://eslint.org/docs/rules/sort-imports

Comment (lkraav) [reply packages/cxl-ui/src/components/cxl-app-layout.js:11]: Yes, I would prefer stylistic changes to be done in a separate commit with a single linter run for everything.

Sort Imports rule makes sense, that could be a separate PR first for `master`, and we'll verify whether everything stays working. There's a couple of places I know that are specific about their import order.

Comment (anoblet) [reply packages/cxl-ui/src/components/cxl-app-layout.js:9]: Fixed

File: packages/cxl-ui/package.json (hunk 3 above)
Comment (anoblet) [reply packages/cxl-ui/package.json:28]: I'm not sure how much I can trim away without there being regressions in certain browsers but I can take a look.

Comment (lkraav) [reply packages/cxl-ui/package.json:28]: Is there a PR we can shoot at normalize-wheel upstream way?

I'm going to publish our current ESM build on our custom registry, but ideally we'd at one point get back to using global NPM for this.

Comment (anoblet) [reply packages/cxl-ui/package.json:28]: I tried to structure it as a PR in the first place. I was unsure if I should use a separate directory under `src` or something else... I was also unsure of what was the best practice for offering CJS/ESM in the same package. Maybe I could leverage the `browser` field in `package.json`?

Comment (anoblet) [reply packages/cxl-ui/package.json:28]: I created a PR here: https://github.com/basilfx/normalize-wheel/pull/3

In the meantime I believe we could rely on this: https://www.npmjs.com/package/normalize-wheel-es

**Comment by lkraav** [comment]
This was deployed live with https://github.com/conversionxl/aybolit/commit/8fd661f6dae993abe4e0b88a7ba935ef68941aea

File: packages/cxl-ui/package.json (hunk 3 above)
Comment (lkraav) [reply packages/cxl-ui/package.json:28]: > https://www.npmjs.com/package/normalize-wheel-es

Looks like they did the same thing, and never bothered to PR.

Let's see what happens in https://github.com/basilfx/normalize-wheel/pull/3 before we make a decision.

Ideally that PR would be a bit less noisy, or code-unified.. but I'm not so well-versed to understand how to pull it off.

**Pull request closed without merging**